src/api/app.py

```diff
@@ -31,24 +66,63 @@
     allow_headers=["*"],
 )
 
+if proxy_target:
+    logging.info(f"Proxy target set to: {proxy_target}")
+    @app.api_route("/{path:path}", methods=["GET", "POST", "PUT", "DELETE", "PATCH", "OPTIONS", "HEAD"])
+    async def proxy(request: Request, path: str):
+        # Build safe target URL
+        path_no_prefix = f"/{path.lstrip('/')}".removeprefix(API_ROUTE_PREFIX)
+        target_url = f"{proxy_target.rstrip('/')}/{path_no_prefix.lstrip('/')}".rstrip("/")
+
+        # Sanitize headers
+        headers = {
+            k: v for k, v in request.headers.items()
+            if k.lower() not in {"host", "content-length", "accept-encoding", "connection"}
+        }
+
+        try:
+            async with httpx.AsyncClient() as client:
+                response = await client.request(
+                    method=request.method,
+                    url=target_url,
+                    headers=headers,
```

Comment:
ideally, we should obtain and pass our own bearer token here, so that the user doesnt have to generate one